handler/authentication.go

```diff
@@ -0,0 +1,224 @@
+	slack        *slack.Client
+}
+
+func NewAuthorizeHandler(
+	e *echo.Echo,
+	logger *zap.Logger,
+	db *gorm.DB,
+	slack *slack.Client,
+) *AuthorizeHandler {
+	clientID := os.Getenv("GITHUB_CLIENT_ID")
+	if len(clientID) == 0 {
+		logger.Fatal("GITHUB_CLIENT_ID is not set")
+	}
+	clientSecret := os.Getenv("GITHUB_CLIENT_SECRET")
+	if len(clientSecret) == 0 {
+		logger.Fatal("GITHUB_CLIENT_SECRET is not set")
+	}
+
+	handler := &AuthorizeHandler{
+		clientID:     os.Getenv("GITHUB_CLIENT_ID"),
+		clientSecret: os.Getenv("GITHUB_CLIENT_SECRET"),
+		logger:       logger,
+		db:           db,
+		slack:        slack,
+	}
+
+	e.GET("/callback/github/auth", handler.handleGithubCallback)
+
+	return handler
+}
+
+type (
+	GetAccessTokenReq struct {
+		ClientID     string `json:"client_id"`
+		ClientSecret string `json:"client_secret"`
+		Code         string `json:"code"`
+		State        string `json:"state"`
+	}
+
+	GetAccessTokenResp struct {
+		AccessToken           string `json:"access_token"`
+		ExpiresIn             int    `json:"expires_in"`
+		RefreshToken          string `json:"refresh_token"`
+		RefreshTokenExpiresIn int    `json:"refresh_token_expires_in"`
+		Scope                 string `json:"scope"`
+		TokenType             string `json:"token_type"`
+	}
+
+	GithubUser struct {
+		Login             string `json:"login"`
+		ID                int    `json:"id"`
+		AvatarURL         string `json:"avatar_url"`
+		GravatarID        string `json:"gravatar_id"`
+		URL               string `json:"url"`
+		HTMLURL           string `json:"html_url"`
+		FollowersURL      string `json:"followers_url"`
+		FollowingURL      string `json:"following_url"`
+		GistsURL          string `json:"gists_url"`
+		StarredURL        string `json:"starred_url"`
+		SubscriptionsURL  string `json:"subscriptions_url"`
+		OrganizationsURL  string `json:"organizations_url"`
+		ReposURL          string `json:"repos_url"`
+		EventsURL         string `json:"events_url"`
+		ReceivedEventsURL string `json:"received_events_url"`
+		Type              string `json:"type"`
+		SiteAdmin         bool   `json:"site_admin"`
+		Name              string `json:"name"`
+		Company           string `json:"company"`
+		Blog              string `json:"blog"`
+		Location          string `json:"location"`
+		Email             string `json:"email"`
+		Hireable          bool   `json:"hireable"`
+		Bio               string `json:"bio"`
+		TwitterUsername   string `json:"twitter_username"`
+		PublicRepos       int    `json:"public_repos"`
+		PublicGists       int    `json:"public_gists"`
+		Followers         int    `json:"followers"`
+		Following         int    `json:"following"`
+		PrivateGists      int    `json:"private_gists"`
+		TotalPrivateRepos int    `json:"total_private_repos"`
+		OwnedPrivateRepos int    `json:"owned_private_repos"`
+		Collaborators     int    `json:"collaborators"`
+	}
+)
+
+func (handler *AuthorizeHandler) handleGithubCallback(ctx echo.Context) error {
```

Comment:
rename `handler` > `h` cho gọn + consistent với các handler khác luôn e

Reply:
Updated that, thank you!